### ISSUE TITLE
Positional-only parameter mistakenly called "keyword parameter" in diagnostic

### ISSUE THREAD

**Issue by debonte** [microsoft/pyright#5032]
**Describe the bug**
Diagnostic refers to a "keyword parameter" that is actually positional-only.

**Code**
```python
from typing import Callable

def called(param: Callable[[], None]):
    pass

def passed(x: str, /):
    pass

called(passed)
```

**To Reproduce**
1. Create file with source code shown above.
2. Note error on argument passed to `called` function on last line.

```text
Argument of type "(x: str, /) -> None" cannot be assigned to parameter "param" of type "() -> None" in function "called"
  Type "(x: str, /) -> None" cannot be assigned to type "() -> None"
    Position-only parameter mismatch; expected 1 but received 0
    Keyword parameter "x" is missing in destination Pyright (reportGeneralTypeIssues)
```

**Expected behavior**
Diagnostic should not refer to `x` as a "Keyword parameter" since `x` is a positional-only parameter.

For what it's worth, I also find the message `Position-only parameter mismatch; expected 1 but received 0` confusing. It seems to me that the "expected" number of arguments should be the number of parameters in the `Callable` parameter (i.e. 0), not the number of parameters in the function that was passed (1). But perhaps that just depends on your point of view.

**VS Code extension or command-line**
Pyright 1.1.305 in VS Code 1.78.0-insider

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.306, which I just published. It will also be included in a future release of pylance.
